This patch is needed *only* for building 1.91.0 with 1.90.0.

Version 1.91.0 uses std lock in bootstrap.  Unfortunately, 1.90.0 does
not support `flock` on Solaris.  As such, we need to supply a
`--bypass-bootstrap-lock` flag, but I cannot sort out exactly how to
specify this in our build process, or if it is even possible.  The
result is this patch, which always excludes bootstrap locking on
Solaris.

Commencing 1.91.0, `flock` is emulated on Solaris via fcntl; see
https://github.com/rust-lang/rust/pull/146269

This patch is no longer needed for 1.91.0 or later.



--- rustc-1.91.0-src/src/bootstrap/src/bin/main.rs.~1~	2025-10-28 16:34:16.000000000 +0000
+++ rustc-1.91.0-src/src/bootstrap/src/bin/main.rs	2025-11-02 17:25:48.792433430 +0000
@@ -40,7 +40,7 @@
 
     let mut build_lock;
 
-    if !config.bypass_bootstrap_lock {
+    if cfg!(not(target_os = "solaris")) && !config.bypass_bootstrap_lock {
         // Display PID of process holding the lock
         // PID will be stored in a lock file
         let lock_path = config.out.join("lock");
